### ISSUE TITLE
Fine-tuning on another dataset

### ISSUE THREAD

**Issue by drake666** [nianticlabs/monodepth2#126]
I'm currently trying to fine-tune monodepth2 on NuScenes. More specifically, I'm fine-tuning the mono_1024x320 model. 

I think I've got everything: 
- I've implemented a new MonoDataset which produce sequential NuScenes images from an index. 
- I've resized (to 570x320) and reflect-padded the image from NuScenes to produce 1024x320 images.
- Ensure that if the frame before or after the requested frame is not in the frame sequence's, the current frame is returned instead (which should produce a nice identity pose and ~0 loss for that data pair).

But after 1 epoch, the model is more or less destroyed and produce weird disparity image (once cropped), like this: 

![image](https://user-images.githubusercontent.com/1728384/71606615-527c4a80-2b40-11ea-95a1-1a0972d99e8a.png)

**Before**

![image](https://user-images.githubusercontent.com/1728384/71606608-3c6e8a00-2b40-11ea-80f5-4100c369f8ee.png)

**After**

So I've got some questions:
1. Should I also fine-tune the pose estimator / pose encoder?
2. Should I revise the way I manage scenes extrema images?
3. Do you think one of the default training parameter is really not appropriate for what I'm trying to do?

I know that the result could also mean just an implementation problem on my part, but the training  and val losses did not looked weird in any way during the training phase (20 epochs), so I think I implemented the NuScenes dataset and its integration into the trainer module correctly.

**Comment by mrharicot** [comment]
Hi,

There's a lot to unpack here!

- The model was trained at the 1024x320 resolution but you can use any resolution close to that one.
I would suggest using a smaller resolution to do initial experiments as it significantly speeds up training. I would use 512x288 or smaller to start your experiments. Do not pad, simply resize. It's alright if the ratio isn't exactly 16x9 as long as the height and width are divisible by 32.
- Reflexion padding a 570 pixels wide image to 1024 won't work as this breaks a lot of assumptions made by the model. Resizing is the best option.
- Discard the first and last frame of each sequence when creating indexes, this will avoid having to pass the same index. It's unclear how the model will react to two identical frames but you can't assume it will "produce a nice identity pose".
- You also need to fine tune the pose model, as it was trained only on kitti.

I hope this helps!

**Comment by drake666** [comment]
It helps a lot, thanks!
I could not hope for a more precise answer :)

**Comment by XXXVincent** [comment]
> It helps a lot, thanks!
> I could not hope for a more precise answer :)

Hello drake666,

Could you share your nuScene data preprocessing scripts?

**Comment by minjabenho** [comment]
Hi there, I am also working on nuScenes data, appreciate if you could share your data processing code with me as well. Many thanks.
